pkg/storage/bucket/s3/config.go

```diff
@@ -95,6 +106,19 @@ func (cfg *Config) Validate() error {
 	return nil
 }
 
+func (cfg *Config) bucketLookupType() s3.BucketLookupType {
```

Comment:
What is your opinion on doing this the same as "BuildThanosConfig()"?

Returning an error when it goes to default which can also return the AutoLookup. The caller can decide what to do. I think at least we are not swallowing any issues.

Reply:
I think `Validate` is sufficient for our case. However, returning an error is better for other use.